catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/main/java/org/codice/ddf/catalog/subscriptionstore/common/SubscriptionPersistor.java

```diff
@@ -0,0 +1,171 @@
+     * <p>
+     * Redundant deletes are valid as a no-op to support javax.cache events propogating write-through
+     * processing to other cache nodes.
+     *
+     * @param subscriptionId the unique id of the subscription to delete.
+     * @throws SubscriptionStoreException if a problem occurs during delete.
+     */
+    public void delete(String subscriptionId) {
+        LOGGER.debug("Deleting [{}] from persistence store. ", subscriptionId);
+        try {
+            persistentStore.delete(PersistentStore.EVENT_SUBSCRIPTIONS_TYPE,
+                    getEcqlStringForId(subscriptionId));
+        } catch (PersistenceException e) {
+            throw new SubscriptionStoreException("Exception while deleting subscription: ", e);
+        }
+    }
+
+    /**
+     * Given subscription metadata, transform it into a persistent item for storage.
+     */
+    private PersistentItem metadataToPersistentItem(SubscriptionMetadata metadata) {
+        notNull(metadata, "subscription metadata cannot be null");
+
+        PersistentItem persistentItem = new PersistentItem();
+        persistentItem.addProperty(SUBSCRIPTION_ID_SOLR, metadata.getId());
+        persistentItem.addProperty(SUBSCRIPTION_TYPE, metadata.getType());
+        persistentItem.addProperty(SUBSCRIPTION_SERIALIZED_FILTER, metadata.getSerializedFilter());
+        persistentItem.addProperty(SUBSCRIPTION_CALLBACK_URL,
+                metadata.getCallbackAddress()
+                        .toString());
+
+        return persistentItem;
+    }
+
+    /**
+     * Given a result from the persistent store, transform the map back into valid subscription
+     * metadata.
+     */
+    private SubscriptionMetadata mapToMetadata(Map<String, Object> map) {
+        notEmpty(map, "persistentItem cannot be null or empty");
+
+        String id = (String) map.get(SUBSCRIPTION_ID_SOLR);
+        String type = (String) map.get(SUBSCRIPTION_TYPE);
+        String serializedSubscription = (String) map.get(SUBSCRIPTION_SERIALIZED_FILTER);
+        String callback = (String) map.get(SUBSCRIPTION_CALLBACK_URL);
+
+        return new SubscriptionMetadata(type, serializedSubscription, callback, id);
+    }
+
+    /**
+     * There may be {@link PersistentStore} implementations that do not use suffix-based key labeling.
+     * This pass-through method allows conditional formating of the retrieved results and tries to
+     * reduce the implementation-dependence of the solution.
+     */
+    private Map<String, Object> formatIfPersistentItem(Map<String, Object> map) {
+        if (map instanceof PersistentItem) {
+            return PersistentItem.stripSuffixes(map);
+        }
+        return map;
+    }
+
+    /**
+     * Generate the ECQL string used for deletes. It will select the entity with the provided
+     * {@code subscriptionId}.
+     */
+    private String getEcqlStringForId(String subscriptionId) {
+        return format("\"%s_txt\" LIKE '%s'", SUBSCRIPTION_ID_SOLR, subscriptionId);
```

Comment:
@paouelle Will add validation to service impl (PR 3/4) to protect against ECQL injection.